zkwrapper/zkwrapper.cc

```diff
@@ -0,0 +1,534 @@
+			prepend_zk_root(path).c_str(),
+			watch,
+			watcherCtx,
+			reinterpret_cast<char *>(data.data()),
+			&len,
+			&stat);
+	if (error_code != ZOK) {
+		LOG(ERROR) << CLASS_NAME <<  "wget on " << path << " failed";
+		print_error(error_code);
+		return false;
+	}
+	data.resize(len);
+	return true;
+}
+
+bool ZKWrapper::get(const std::string &path,
+		std::vector <std::uint8_t> &data,
+		int &error_code) const {
+
+	// TODO: Make this a constant value. Define a smarter retry policy for oversized data
+	struct Stat stat;
+	int len = MAX_PAYLOAD;
+	// TODO: Perhaps we can be smarter about this
+	// LOG(INFO) << CLASS_NAME <<  "Data resizing to " << len;
+	data.resize(len);
+	// LOG(INFO) << CLASS_NAME <<  "Data resizing to 1;" << data.size();
+	error_code = zoo_get(zh,
+			prepend_zk_root(path).c_str(),
+			0,
+			reinterpret_cast<char *>(data.data()),
+			&len,
+			&stat);
+	if (error_code != ZOK) {
+		LOG(ERROR) << CLASS_NAME <<  "get on " << path << " failed";
+		print_error(error_code);
+		return false;
+	}
+	data.resize(len);
+	return true;
+}
+
+bool ZKWrapper::set(const std::string &path,
+		const std::vector <std::uint8_t> &data,
+		int &error_code,
+		int version) const {
+
+	error_code = zoo_set(zh,
+			prepend_zk_root(path).c_str(),
+			reinterpret_cast<const char *>(data.data()),
+			data.size(),
+			version);
+	if (error_code != ZOK) {
+		LOG(ERROR) << CLASS_NAME <<  "set on " << path << " failed";
+		print_error(error_code);
+		return false;
+	}
+	return true;
+}
+
+bool ZKWrapper::exists(const std::string &path,
+		bool &exist,
+		int &error_code) const {
+	// TODO: for now watch argument is set to 0, need more error checking
+	int rc = zoo_exists(zh, prepend_zk_root(path).c_str(), 0, 0);
+	error_code = rc;
+	if (rc == ZOK) {
+		exist = true;
+		return true;
+	} else if (rc == ZNONODE) {
+		exist = false;
+		return true;
+	} else {
+		// NOTE: value exist is not updated in this case
+		LOG(ERROR) << CLASS_NAME <<  "exists on " << path << " failed";
+		print_error(error_code);
+		return false;
+	}
+}
+
+bool ZKWrapper::wexists(const std::string &path,
+		bool &exist,
+		watcher_fn watch,
+		void *watcherCtx,
+		int &error_code) const {
+	struct Stat stat;
+	int rc = zoo_wexists(zh, prepend_zk_root(path).c_str(), watch, watcherCtx, &stat);
+	error_code = rc;
+	if (rc == ZOK) {
+		exist = true;
+		return true;
+	} else if (rc == ZNONODE) {
+		exist = false;
+		return true;
+	} else {
+		// NOTE: value exist is not updated in this case
+		LOG(ERROR) << CLASS_NAME <<  "wexists on " << path << " failed";
+		print_error(error_code);
+		return false;
+	}
+}
+
+bool ZKWrapper::delete_node(const std::string &path, int &error_code) const {
+	// NOTE: use -1 for version, check will not take place.
+	error_code = zoo_delete(zh, prepend_zk_root(path).c_str(), -1);
+	if (error_code != ZOK) {
+		LOG(ERROR) << CLASS_NAME <<  "delete on " << path << " failed";
+		print_error(error_code);
+		return false;
+	}
+	return true;
+}
+
+// TODO: Modify
```

Comment:
Huh? Please write better TODOs. Gives no information about what needs to be done.